File: Codility/Lessons/02. Arrays/CyclicRotation/solution.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def solution(a, k)
  # write your code in Ruby 2.2
  rotated = Array.new a.length

  a.each_with_index do |x, index|
    new_index = index + k
    if new_index > (a.length - 1)
      new_index = new_index - a.length
    end

    rotated[new_index] = x
  end

  rotated
end

File: Codility/Lessons/02. Arrays/CyclicRotation/solution.go
package solution

// you can also use imports, for example:
// import "fmt"

// you can write to stdout for debugging purposes, e.g.
// fmt.Println("this is a debug message")

func Solution(A []int, K int) []int {
    // write your code in Go 1.4
    if len(A) == 0 {
        return A
    }

    if K >= len(A) {
        K = K % len(A)
    }

    var rotated = make([]int, len(A))
    for index, x := range A {
        new_index := index + K

        if new_index > (len(A) - 1) {
            new_index = new_index - len(A)
        }

        rotated[new_index] = x
    }

    return rotated
}
